### PR TITLE
Better respect lang version when generating expression bodies.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/20352

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @sharwell @Pilchie

**Comment by CyrusNajmabadi** [comment]
Previously, we only supported 'use expression body' for C# 7 and above.  That's the version of C# that has expression bodies supported for all members.  Now we support it as well for C#6 if it's for one of the constructs (like methods/operators) that supported it back then.

File: src/EditorFeatures/CSharpTest/UseExpressionBody/Analyzer/UseExpressionBodyForMethodsAnalyzerTests.cs [hunk 1]
```diff
@@ -427,6 +427,52 @@ void M()
...
+using System;
+class C
+{
+    void M() [|=>|] 0;
+}", new TestParameters(options: UseExpressionBody, parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp6)));
+        }
+
+        [WorkItem(20352, "https://github.com/dotnet/roslyn/issues/20352")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseExpressionBody)]
+        public async Task TestOfferToConvertToExpressionIfCSharp6()
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/UseExpressionBody/Analyzer/UseExpressionBodyForMethodsAnalyzerTests.cs]: :bulb: Can you add an additional test `TestDoNotOfferToConvertToExpressionIfCSharp5`?

File: src/EditorFeatures/CSharpTest/UseExpressionBody/Analyzer/UseExpressionBodyForMethodsAnalyzerTests.cs [hunk 2]
```diff
@@ -427,6 +427,52 @@ void M()
 }", options: UseExpressionBody, parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp5));
         }
 
+        [WorkItem(20352, "https://github.com/dotnet/roslyn/issues/20352")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseExpressionBody)]
+        public async Task TestDoNotOfferToConvertToBlockIfExpressionBodyPreferredIfCSharp6()
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/UseExpressionBody/Analyzer/UseExpressionBodyForMethodsAnalyzerTests.cs]: :bulb: Can you add an additional test `TestOfferToConvertToBlockIfExpressionBodyPreferredIfCSharp5`?

Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/UseExpressionBody/Analyzer/UseExpressionBodyForMethodsAnalyzerTests.cs]: Already have that one.  It's called TestOfferToConvertToBlockEvenIfExpressionBodyPreferredIfPriorToCSharp6  (yes, my naming isn't terrific) :)

**Pull request merged**